### ISSUE TITLE
m2ts files fail to encode

### ISSUE THREAD

**Issue by MaciejGorczyca** [master-of-zen/Av1an#147]
I tried to encode m2ts files with av1an, which resulted in an error:

```
$ av1an -p 1 -i 00006.m2ts -enc rav1e -v " --quantizer 150 -s 10 " -o output/rav1e10_150
100%|##########| 408/408 [00:02<00:00, 161.08frames/s]
Traceback (most recent call last):
  File "c:\program files\python\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\program files\python\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Program Files\Python\Scripts\av1an.exe\__main__.py", line 7, in <module>
  File "c:\program files\python\lib\site-packages\av1an.py", line 18, in main
    Av1an().main_thread()
  File "c:\program files\python\lib\site-packages\av1an.py", line 14, in main_thread
    main_queue(self.args)
  File "c:\program files\python\lib\site-packages\Av1an\encode.py", line 45, in main_queue
    encode_file(args)
  File "c:\program files\python\lib\site-packages\Av1an\encode.py", line 78, in encode_file
    chunk_queue = load_or_gen_chunk_queue(args, resuming, split_locations)
  File "c:\program files\python\lib\site-packages\Av1an\chunk_queue.py", line 60, in load_or_gen_chunk_queue
    chunk_queue = create_encoding_queue(args, split_locations)
  File "c:\program files\python\lib\site-packages\Av1an\chunk_queue.py", line 80, in create_encoding_queue
    chunk_queue = chunk_method_gen[args.chunk_method](args, split_locations)
  File "c:\program files\python\lib\site-packages\Av1an\chunk_queue.py", line 113, in create_video_queue_hybrid
    to_split = reduce_segments(to_split)
  File "c:\program files\python\lib\site-packages\Av1an\chunk_queue.py", line 91, in reduce_segments
    scenes = scenes[::interval]
ValueError: slice step cannot be zero
```

The same ffmpeg command encodes just fine: `ffmpeg -y -i 00006.m2ts -c:v librav1e -strict experimental -speed 10 -qp 150 output2.mkv`

Replacing m2ts file for a different mov file causes av1an to work just fine.

Here is the m2ts file used in this case: https://drive.google.com/file/d/1GG14JWfKCAlRgeewEPDGnqEw5qsvf2YM/view

**Comment by master-of-zen** [comment]
@MaciejGorczyca ffmpeg returns me that error if I try to copy that to another container
`[matroska @ 0x557a56aa2500] No wav codec tag found for codec pcm_bluray
Could not write header for output file #0 (incorrect codec parameters ?): Invalid argument`

**Comment by Alkl58** [comment]
ffmpeg afaik can't handle the pcm audio streams from bluray.
reencoding the audio to, for example, pcm_s16le will solve this.

**Comment by MaciejGorczyca** [comment]
For me, ffmpeg automatically changed pcm_bluray to vorbis and successfully encoded the file:

```
Stream mapping:
  Stream #0:0 -> #0:0 (h264 (native) -> av1 (librav1e))
  Stream #0:1 -> #0:1 (pcm_bluray (native) -> vorbis (libvorbis))
```

I assume av1an tries to copy audio stream and that's why it fails. Something that could be kept in mind for future. I removed audio stream as it is unnecessary for my case (and I can add the stream later on anyway).

I also located the issue. The chunk_queue.py tries to look for keyframes and check them against splits:

```python
    keyframes = get_keyframes(args.input)

    end = [frame_probe_cv2(args.input)] if frame_probe_cv2(args.input) > 1 else [frame_probe(args.input)]

    splits = [0] + split_locations + end

    segments_list = list(zip(splits, splits[1:]))
    to_split = [x for x in keyframes if x in splits]
```

The problem is that there are no keyframes but 13 splits. The resulting list is 0 splits and causes the app to crash. I solved it temporarily by replacing `to_split = [x for x in keyframes if x in splits]` to `to_split = splits`

I don't know why the keyframes are checked against splits and didn't try to get into what works where and how, so all I can do is provide insight. You guys decide what steps to take.
